### PR TITLE
Display 2D boundary colored by attribute

### PR DESCRIPTION
**Pull request by tzanio**
See https://github.com/GLVis/glvis/issues/204

### CODE REVIEW

**Comment by tzanio** [comment]
As an example, the output of 

```
./glvis -m ../mfem/data/beam-quad.mesh -mslw 6 -oldgl -k "ebbAPPPP/////////c"
```

in this PR looks like this:

Screen Shot 2021-10-31 at 11 54 16 AM

**Review by carlos-brito-pacheco — APPROVED** [review]
Hi @tzanio . Thank you for adding me to this review and implementing this! Just to let you know this is working fine for me on my machine and my use case. Also the code looks fine to me. Thanks again!

File: README [hunk 1]
```diff
@@ -125,8 +125,11 @@ O - Cycle through the "subdivision functions", (prints a message in the
 o - perform the current "subdivision function"
 
 A - Turn on/off the use of anti-aliasing/multi-sampling
-b - Display/Hide the boundary in 2D scalar mode. Use Shift+F9/F10 to cycle
-    through the boundary attributes.
+b - Toggle the boundary in 2D scalar mode.
+    The options are: -> no boundary
+                     -> black boundary
+		     -> boundary colored with the boundary attribute
```
Comment (v-dobrev) [inline README:131]:
```suggestion
                     -> black boundary
                     -> boundary colored with the boundary attribute
```

File: lib/vssolution.hpp [hunk 2]
```diff
@@ -38,12 +38,15 @@ class VisualizationSceneSolution : public VisualizationSceneScalarData
 
    gl3::GlDrawable lcurve_buf;
    gl3::GlDrawable line_buf;
-   gl3::GlDrawable bdr_buf;
+   gl3::GlDrawable bdr_buf1;
+   gl3::GlDrawable bdr_buf2;
```
Comment (v-dobrev) [inline lib/vssolution.hpp:42]: Why use two boundary buffers? Just Call `PrepareBoundary()` when toggling `drawbdr` to 1 or 2.

**Comment by v-dobrev** [comment]
See proposed tweaks in #208.

**Comment by tzanio** [comment]
@v-dobrev and @pazner -- this is good to go IMHO.

File: CHANGELOG [hunk 3]
```diff
@@ -9,6 +9,15 @@
                               http://glvis.org
 
 
+Version 4.1.1 (development)
+===========================
+
+- Added a third mode to keys 'b'/'B' in 2D to display the mesh boundary colored
+  by boundary element attribute.
+
+- Save and restore the value range for the solution when using keys 'e' and 'b'.
```
Comment (v-dobrev) [inline CHANGELOG:18]:
```suggestion
- In 2D, save and restore the value range for the solution when using keys 'e' and 'b'.
```

**Review by v-dobrev — APPROVED** [review]

**Review by pazner — APPROVED** [review]
Great addition, it will be nice to have this natively in GLVis!

**Pull request merged**